File: frontend/src/types/enums.ts
/**
 * @file
 * Defines domain enums mirroring the backend's core enums.
 */
export const TaskStatus = {
    ACCEPTED: "accepted",
    PROCESSING: "processing",
    COMPLETED: "completed",
    ERROR: "error",
} as const;

export type TaskStatus = typeof TaskStatus[keyof typeof TaskStatus];

export const AuditIssueType = {
    GHOST: "ghost",
    ORPHAN: "orphan",
} as const;

export type AuditIssueType = typeof AuditIssueType[keyof typeof AuditIssueType];

export const AuditIssueStatus = {
    PENDING: "pending",
    RESOLVED: "resolved",
    IGNORED: "ignored",
} as const;

export type AuditIssueStatus = typeof AuditIssueStatus[keyof typeof AuditIssueStatus];

export const ImageRating = {
    SAFE: "safe",
    QUESTIONABLE: "questionable",
    EXPLICIT: "explicit",
} as const;

export type ImageRating = typeof ImageRating[keyof typeof ImageRating];

export const BulkOperationMode = {
    REPLACE: "replace",
    APPEND: "append",
    REMOVE: "remove",
} as const;

export type BulkOperationMode = typeof BulkOperationMode[keyof typeof BulkOperationMode];

export const CREATOR_TYPES = ['Artist', 'AI Generated', 'Studio', 'Photography', 'Cosplayer', 'Model', 'Unknown'] as const;

File: backend/app/core/enums.py
"""Centralized enums for domain constants used across the application."""

from enum import StrEnum


class TaskStatus(StrEnum):
    ACCEPTED = "accepted"
    PROCESSING = "processing"
    COMPLETED = "completed"
    ERROR = "error"


class AuditIssueType(StrEnum):
    GHOST = "ghost"
    ORPHAN = "orphan"
    DUPLICATE_ENTRY = "duplicate_entry"
    EMPTY_SET = "empty_set"
    GHOST_SET = "ghost_set"
    CORRUPTED_IMAGE = "corrupted_image"
    PATH_MISMATCH = "path_mismatch"
    ORPHAN_TAG = "orphan_tag"
    ORPHAN_CREATOR = "orphan_creator"
    ORPHAN_CHARACTER = "orphan_character"


class AuditIssueStatus(StrEnum):
    PENDING = "pending"
    RESOLVED = "resolved"
    IGNORED = "ignored"


class ImageRating(StrEnum):
    SAFE = "safe"
    QUESTIONABLE = "questionable"
    EXPLICIT = "explicit"


class BulkOperationMode(StrEnum):
    REPLACE = "replace"
    APPEND = "append"
    REMOVE = "remove"
